### PR TITLE
NH064 - add Dockerfile to the service repository

### PR DESCRIPTION
**Pull request by lukaswajdzik**
Implementation of https://github.com/nhood-org/nhood-docs/issues/64

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM maven:3.6-jdk-11
+RUN mkdir -p /usr/src/input
```
Comment (krzysztofreczek) [inline Dockerfile]: I believe we do not need those directories anymore. I would keep just a simple `/output` directory as an interface between docker containers.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+RUN mkdir -p /usr/src/output  
+USER root
```
Comment (krzysztofreczek) [inline Dockerfile]: Do we really need root here? Let's simplify this docker to the minimum if possible

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+WORKDIR /usr/src/input
+ADD . /usr/src/input/app
+RUN cd ./app && \
+    mvn clean install
+RUN JAR_FILE=$(find . -name 'nhood*app*.jar') && \
```
Comment (krzysztofreczek) [inline Dockerfile]: Do find command within `./nhood-data-url-svc-app/target` directory only. There may be multiple jar files build in the project and this solution my be error-prone in the future.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    cp $JAR_FILE /usr/src/output/app.jar
+
+FROM openjdk:11
+COPY --from=0 /usr/src/output/app.jar app.jar
+EXPOSE 8080
```
Comment (krzysztofreczek) [inline Dockerfile]: this is not right (I know the bug was already there:)). It should be EXPOSE $PORT if such construction is possible. If not, remove it.

File: README.md [hunk 5]
```diff
@@ -65,6 +65,15 @@ mvn spring-boot:run -Dspring-boot.run.profiles=local -f nhood-data-url-svc-app/p
 
 Open [http://localhost:8080/swagger-ui.html](http://localhost:8080/swagger-ui.html)
 
+
+## Build docker image with the application
+
+In order to get docker image deployed from the local repository version, use the following command:
```
Comment (krzysztofreczek) [inline README.md]: 👍

**Comment by lukaswajdzik** [comment]
my local test run with EXPORT $PORT:

Step 9/10 : EXPOSE $PORT
 ---> Running in 47bea0032e60
Removing intermediate container 47bea0032e60
 ---> c30779db4f56
Step 10/10 : CMD java     -Djava.security.egd=file:/dev/./urandom     -Dspring.profiles.active=prod     -Dserver.port=$PORT     -jar app.jar
 ---> Running in f6f9cb7255cb
Removing intermediate container f6f9cb7255cb
 ---> 175d5bbab075
**Successfully** built 175d5bbab075
**Successfully** tagged nhood/nhood-data-url-svc:latest

but, circleci not

**Comment by krzysztofreczek** [comment]
```
Step 9/10 : EXPOSE $PORT
EXPOSE requires at least one argument
```

Looks like env variable $PORT is not available during build execution. I think this may help here: https://circleci.com/docs/2.0/env-vars/ Just set PORT=8080 for testing purposes

File: .circleci/config.yml [hunk 6]
```diff
@@ -30,11 +30,18 @@ jobs:
...
+    steps:
+      - checkout
+      - run: docker build -t nhood/nhood-data-url-svc:test .
+      
 workflows:
   version: 2
   build-deploy:
     jobs:
       - build
+      - docker_image
```
Comment (krzysztofreczek) [inline .circleci/config.yml]: change this name to `build_docker`. I really like common prefixes as those groups items nicely.

File: .circleci/config.yml [hunk 7]
```diff
@@ -15,6 +15,13 @@ jobs:
     - run: mvn clean install
     - store_test_results:
         path: target/surefire-reports
+
+  build_docker:
+    machine: true
+    steps:
+      - checkout
+      - run:
+          command: docker build --build-arg PORT=8080 -t nhood/nhood-data-url-svc:test .
```
Comment (lukaswajdzik) [inline .circleci/config.yml]: _command:_ is not needed here. 
It could be used for multi line commands running in one environment

Comment (lukaswajdzik) [reply .circleci/config.yml]: Fix in the next commit

**Pull request merged**